subsys/bluetooth/mesh/vnd/le_pair_resp.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/random/rand32.h>
+
+#include <zephyr/bluetooth/conn.h>
+#include <bluetooth/mesh/vnd/le_pair_resp.h>
+#include "../model_utils.h"
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+#define LOG_LEVEL CONFIG_BT_MESH_MODEL_LOG_LEVEL
+#include "zephyr/logging/log.h"
+LOG_MODULE_REGISTER(bt_mesh_le_pair_resp);
+
+#define BT_MESH_LE_PAIR_OP BT_MESH_MODEL_OP_3(0x11, BT_MESH_VENDOR_COMPANY_ID_LE_PAIR_RESP)
+
+#define BT_MESH_LE_PAIR_OP_RESET 0x00
+#define BT_MESH_LE_PAIR_OP_STATUS 0x01
+
+#define STATUS_PASSKEY_SET 0x00
+#define STATUS_PASSKEY_NOT_SET 0x01
+
+static int handle_reset(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
```

Comment:
We discussed that we should give ability for remote user to specify certain passkey over the model message.

I would imaging 'RESET' message to contain optional 'PASSKEY' field. If this field is present, the passkey contained in the field is used, otherwise, randomly generated passkey used.

If this seems unsecure, it could also be seperate sub-opcode to set a given passkey for debugging/developmental purposes and this subopcode could then be compiled-out if not needed.

Reply:
I'm not sure a remote user needs this. If we are talking about mobile apps, this makes no sense as you can't set passkey for a pairing request on a phone without a physical user action. In case of embedded device as initiator, if it can send a message then it can also receive a message, so it is not a problem for the embedded initiator to extract the passkey from the Status message. I left ability for an application to hardcode a passkey if they need.

Reply:
I feel hardcoding passkey in the application is bit risky if someone forgets to remove that from the production version. Whereas, specifying static passkey via mobile app (or tester) would then be limited to only local setup without having to modify anything in the production version for the purpose of bench testing.

Reply:
> I feel hardcoding passkey in the application is bit risky if someone forgets to remove that from the production version.

The same can happen if the passkey is passed over a message.

> Well, mostly I see that, they can also modify this vendor model by themselves and add another message to do this, however, that won't be supported in nRF Mesh mobile app out of the box, and they will have to do something more there.

Same as the comment above. They can forget removing this from production code.

> the UX will be cumbersome for developers.

It will always be cumbersome as as I mentioned above, the mobile phone always requires a user action. Hardcoding the passkey IMHO also makes developers skip the important part of the process of showing the passkey before the pairing. Remember that here the passkey can be generated at any time before the next pairing.

> To avoid this, I would still recommend to add option to get passkey from mobile app and then use that. This could probably be done later as well as long as we are okay to modify model messages later.

A new message (e.g. Passkey Set) can be added later as well. No need to pile up all possible behaviors in a single message.

Reply:
I disagree with this part:

> It will always be cumbersome as as I mentioned above, the mobile phone always requires a user action. Hardcoding the passkey IMHO also makes developers skip the important part of the process of showing the passkey before the pairing. Remember that here the passkey can be generated at any time before the next pairing.

Imagine: you have a Nordic or fork-of-Nordic nRF Mesh app 
=> App shows the text field to input desired passkey to be set by embedded device 
=> When app boots app, the field is blank, to indicate the app-library to send PassKey_Reset message
=> When developers or network deployers are testing, or when they want to have ease of installation, on-purpose they set the desired passkey '123456' in the text field, so that they don't have to remember randomly generated 6 digit key every time they test or send image to new devices under test.
=> This instructs library to include optional passkey field in PassKey_Reset message. As field is present, embedded side uses this instead of random passkey.
=> The mobile app when shipped, does not hardcode anything and allows end user to choose.

If someone forgets to remove this from mobile app, the app is easy to update (than someone forgetting to removed hardcoded passkey from embedded firmware, and then have to push DFU to fix it). This mechanism gives a good UX for developer IMO by removing a need to memorize a new random number every time they test DFU image upload.

Wouldn't it?

Reply:
> If someone forgets to remove this from mobile app, the app is easy to update (than someone forgetting to removed hardcoded passkey from embedded firmware, and then have to push DFU to fix it). 

As well as there are many other places to shoot themselves in the foot, not only this one. And if we _really_ want to avoid customers shooting themselves into the foot, we should not allow to set the passkey over the protocol in the first place.